src/Net/Http/HttpOptions.php

```diff
@@ -18,6 +24,14 @@ public function __construct()
     }
 
     /**
+     * @return SslOptions
+     */
+    public function getSslOptions()
+    {
+        return $this->sslOptions ?: $this->sslOptions = new SslOptions;
```

Comment:
Although not strictly a getter this is just a lazy loading pattern which is an internal optimization the caller does not need to be aware of. We could rename the method *getOrCreateSslOptions*, but again, the caller doesn't need to be aware of this implementation detail. If we later remove the optimization and always `SslOptions` in the constructor this would require another name change and yet, from the client perspective, nothing is different. Ergo the name is correct.